tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/sr_evaluator.py

```diff
@@ -0,0 +1,572 @@
+            self.output_blob = output_blob
+            self.with_prefix = with_prefix
+
+    def load_model(self, network_info, launcher, log=False):
+        if 'onnx_model' in network_info:
+            network_info.update(launcher.config)
+            model, weights = launcher.convert_model(network_info)
+        else:
+            model, weights = self.automatic_model_search(network_info)
+        if weights is not None:
+            self.network = launcher.read_network(str(model), str(weights))
+            self.exec_network = launcher.ie_core.load_network(self.network, launcher.device)
+        else:
+            self.exec_network = launcher.ie_core.import_network(str(model))
+        self.set_input_and_output()
+        if log:
+            self.print_input_output_info()
+
+
+def create_model(model_config, launcher, delayed_model_loading=False):
+    launcher_model_mapping = {
+        'dlsdk': ModelDLSDKModel,
+        'tf': ModelTFModel,
+        'dummy': DummyModel
+    }
+    framework = launcher.config['framework']
+    if 'predictions' in model_config and not model_config.get('store_predictions', False):
+        framework = 'dummy'
+    model_class = launcher_model_mapping.get(framework)
+    if not model_class:
+        raise ValueError('model for framework {} is not supported'.format(framework))
+    return model_class(model_config, launcher, delayed_model_loading)
+
+
+class SRFModel(BaseModel):
+    def __init__(self, network_info, launcher, models_args, is_blob, delayed_model_loading=False):
+        super().__init__(network_info, launcher)
+        if models_args and not delayed_model_loading:
+            model = network_info.get('model', {})
+            if not contains_any(model, ['model', 'onnx_model']) and models_args:
+                model['model'] = models_args[0]
+                model['_model_is_blob'] = is_blob
+            network_info.update({'model': model,})
+        if not contains_all(network_info, ['model',]) and not delayed_model_loading:
+            raise ConfigError('network_info should contain encoder and decoder fields')
+        # self.num_processing_frames = network_info['decoder'].get('num_processing_frames', 16)
+        # self.processing_frames_buffer = []
+        self.model = create_model(network_info['model'], launcher, delayed_model_loading)
+        self.feedback = self.model.feedback
+        self.init_feedback = self.model.init_feedback
+        self.fill_feedback = self.model.fill_feedback
+        self.store_predictions = network_info['model'].get('store_predictions', False)
+        self._predictions = [] if self.store_predictions else None
+        self._part_by_name = {'model': self.model}
+        self._raw_outs = OrderedDict()
+
+    def predict(self, identifiers, input_data, callback=None):
+        predictions, raw_outputs = [], []
+        for data in input_data:
+            output, prediction = self.model.predict(identifiers, data)
+            if callback:
```

Comment:
no need in callback here, if you have only single model. You should provide model raw outputs with metric result in evaluator instead of providing them separated or twice

Reply:
fixed